torch/_inductor/fx_passes/quantization.py

```diff
@@ -2607,16 +2610,29 @@ def get_pattern_no_bias(expand_a_scale: bool):
         KeywordArg("out_shape_with_bias"),
     )
 
+    # The following patterns are for torchao int8_dynamic_activation_int8_weight linear,
+    # when both activation and weights are symmetrically quantized.
+    # In practice, though, they may also match smooth-quant pattern when a 2D input shape would be used.
+    # Since add is not currently being used as a oneDNN post-op, but is unfused, we don't need these patterns with bias.
+    # Ideally, we should add mul + add post-op support in ATen int8 oneDNN linear op.
+    pattern1_with_no_outer_or_act_reshape = get_pattern_no_bias(
```

Comment:
I have a question regarding to the bias. In @Xia-Weiwen's base PR, it handles 2 cases:

- Case 1: when activation is per-tensor quant, the bias can be one of the inputs to qlinear.
- Case 2: when activation is per-channel quant, the bias can't be fused and will exist as epilogue.

But here in this PR, we only register the pattern wo bias and may cause the difference of case 1. May I know the reason of this difference?

Reply:
> But here in this PR, we only register the pattern wo bias and may cause the difference of case 1. May I know the reason of this difference?

`int8_dynamic_activation_int8_weight` supports per-token quantization of activation, but not a scalar activation scale.

Reply:
Basically, in case of `int8_dynamic_activation_int8_weight`, the activation scale is a vector.

@leslie-fang-intel, if the case of `smooth-quant` with 2D activation & scalar activation scale is to be supported, then bias would also have to be in the pattern. Please let me know if its support also needs to be added.